x/stakeibc/types/message_create_trade_route_test.go

```diff
@@ -0,0 +1,169 @@
+
+	invalidMessage = validMessage
+	invalidMessage.StrideToTradeConnectionId = "connection-X"
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid stride to trade connection ID")
+
+	// Set invalid channel IDs - confirm invalid
+	invalidMessage = validMessage
+	invalidMessage.HostToRewardTransferChannelId = ""
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid host to reward channel ID")
+
+	invalidMessage = validMessage
+	invalidMessage.RewardToTradeTransferChannelId = "channel-"
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid reward to trade channel ID")
+
+	invalidMessage = validMessage
+	invalidMessage.TradeToHostTransferChannelId = "channel-X"
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid trade to host channel ID")
+
+	// Set invalid denom's - confirm invalid
+	invalidMessage = validMessage
+	invalidMessage.RewardDenomOnHost = "not-ibc-denom"
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid reward denom on host")
+
+	invalidMessage = validMessage
+	invalidMessage.RewardDenomOnReward = ""
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid reward denom on reward")
+
+	invalidMessage = validMessage
+	invalidMessage.RewardDenomOnTrade = "not-ibc-denom"
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid reward denom on trade")
+
+	invalidMessage = validMessage
+	invalidMessage.HostDenomOnTrade = "not-ibc-denom"
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid host denom on trade")
+
+	invalidMessage = validMessage
+	invalidMessage.HostDenomOnHost = ""
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid host denom on host")
+
+	// Set invalid pool configurations - confirm invalid
+	invalidMessage = validMessage
+	invalidMessage.PoolId = 0
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "invalid pool id")
+
+	invalidMessage = validMessage
+	invalidMessage.MinSwapAmount = sdkmath.NewInt(10)
+	invalidMessage.MaxSwapAmount = sdkmath.NewInt(9)
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "min swap amount cannot be greater than max swap amount")
+
+	invalidMessage = validMessage
+	invalidMessage.MaxAllowedSwapLossRate = ""
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "unable to cast max allowed swap loss rate to a decimal")
+
+	invalidMessage = validMessage
+	invalidMessage.MaxAllowedSwapLossRate = "-0.01"
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "max allowed swap loss rate must be between 0 and 1")
+
+	invalidMessage = validMessage
+	invalidMessage.MaxAllowedSwapLossRate = "1.01"
+	require.ErrorContains(t, invalidMessage.ValidateBasic(), "max allowed swap loss rate must be between 0 and 1")
+}
+
+func TestValidateConnectionId(t *testing.T) {
+	require.NoError(t, types.ValidateConnectionId("connection-0"))
+	require.NoError(t, types.ValidateConnectionId("connection-10"))
+	require.NoError(t, types.ValidateConnectionId("connection-1203"))
```

Comment:
I was wondering that too, and I'm not sure, but I borrowed the regex from ibc so I think it's probably legal

Really this check is just a guardrail to catch a fat finger error earlier if possible. If the connection doesn't exist, it will error when it actually executes